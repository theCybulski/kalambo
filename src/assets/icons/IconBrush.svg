<svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0)">
<path d="M21.8156 27.2906V23.1375C21.8156 22.6687 21.4312 22.2844 20.9625 22.2844H9.03745C8.5687 22.2844 8.18433 22.6687 8.18433 23.1375V27.2906" stroke="#323232" stroke-width="2" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M19.2469 18.7031C19.0688 11.6156 16.275 7.63124 15 5.78436C13.725 7.63124 10.9313 11.6062 10.7532 18.7031H19.2469Z" stroke="#323232" stroke-width="2" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M15 0.9375C7.24687 0.9375 0.9375 7.24687 0.9375 15C0.9375 22.7531 7.24687 29.0625 15 29.0625C22.7531 29.0625 29.0625 22.7531 29.0625 15C29.0625 7.24687 22.7531 0.9375 15 0.9375Z" stroke="#323232" stroke-width="2" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="30" height="30" fill="white"/>
</clipPath>
</defs>
</svg>
