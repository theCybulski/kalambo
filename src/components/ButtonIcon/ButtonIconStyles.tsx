import styled, { css } from 'styled-components';

export const Icon = styled.div`
  ${({ theme, icon, isActive }) => css`
    content: '';
    display: block;
    width: 30px;
    height: 30px;
    mask-image: url(${icon});
    mask-size: contain;
    background-color: rgba(${theme.colors.additional}, 1);
    opacity: 0.5;
    transition: opacity 0.2s ease-out;

    ${isActive &&
    css`
      opacity: 1;
      background-color: rgba(${theme.colors.secondaryColor}, 1);
    `}
  `}
`;

export const Label = styled.span`
  ${({ theme, isActive }) => css`
    margin-left: 10px;
    font-size: ${theme.colors.additional};
    opacity: 0.6;

    ${isActive &&
    css`
      opacity: 1;
    `}
  `}
`;

export const Wrapper = styled.button`
  position: relative;
  background-color: transparent;
  border: none;
  padding: 0;
  cursor: pointer;
  display: flex;
  align-items: center;

  &:hover {
    ${Icon} {
      opacity: 1;
    }

    ${Label} {
      opacity: 1;
    }
  }
`;
