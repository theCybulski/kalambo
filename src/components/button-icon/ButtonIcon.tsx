import React, { ButtonHTMLAttributes } from "react";
import cn from "classnames";

import styles from "./ButtonIcon.module.scss";

export enum BUTTON_ICON_VARIANTS {
  BRUSH = "brush",
  ERASER = "eraser",
  CLEAR = "clear"
}

export interface ButtonIconProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isActive?: boolean;
  variant: BUTTON_ICON_VARIANTS
}

export const ButtonIcon: React.FC<ButtonIconProps> = ({ variant, isActive, children, ...props }) => {
  return (
    <button
      className={cn(
        styles.wrapper,
        { [styles.isActive]: isActive }
      )}
      {...props}
    >
      <div className={cn(styles.icon, styles[variant])} />
      {children && <span>{children}</span>}
    </button>
  );
};
