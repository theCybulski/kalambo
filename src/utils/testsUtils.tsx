import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from 'shared/styles/defaultTheme';
import lightTheme from 'shared/styles/lightTheme';

export const mountWithTheme = (component) =>
  mount(<ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>{component}</ThemeProvider>);
