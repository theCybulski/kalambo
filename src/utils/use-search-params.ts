import { useLocation } from 'react-router-dom';

export const useSearchParams = () => {
  const { search } = useLocation();

  const params = new URLSearchParams(search);

  let paramsObj: { [key: string]: string } = {};
  params.forEach((value, key) =>
    paramsObj = { ...paramsObj, [key]: value });

  return paramsObj;
};
