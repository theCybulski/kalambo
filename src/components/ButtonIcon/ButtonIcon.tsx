import React, { ButtonHTMLAttributes } from 'react';

import * as Styled from './ButtonIconStyles';

export interface ButtonIconProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  icon: string;
  isActive?: boolean;
}

const ButtonIcon: React.FC<ButtonIconProps> = ({ icon, isActive, children, ...props }) => (
  <Styled.Wrapper {...props}>
    <Styled.Icon icon={icon} isActive={isActive} />
    {children && <Styled.Label isActive={isActive}>{children}</Styled.Label>}
  </Styled.Wrapper>
);

export default ButtonIcon;
