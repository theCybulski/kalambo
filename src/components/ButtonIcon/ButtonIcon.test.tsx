import React from 'react';
import toJson from 'enzyme-to-json';
import { mountWithTheme } from 'utils/testsUtils';

import ButtonIcon from './ButtonIcon';
import IconBrush from 'assets/icons/IconBrush.svg';

describe(`${ButtonIcon.name}`, () => {
  let context: any;
  let component: any;

  beforeEach(() => {
    context = {
      defaultProps: {
        icon: IconBrush,
        isActive: false,
        onClick: jest.fn(),
      },
    };

    component = mountWithTheme(<ButtonIcon {...context.defaultProps} />).find(ButtonIcon);
  });

  it('Renders properly with default props', () => {
    expect(component.props().isActive).toBe(false);
    expect(component.props().icon).toBe(IconBrush);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('Accepts children', () => {
    const component2 = mountWithTheme(
      <ButtonIcon {...context.defaultProps}>Some label</ButtonIcon>
    ).find(ButtonIcon);

    expect(component2.text()).toBe('Some label');
  });

  it('Calls onClick event', () => {
    component.simulate('click');

    expect(context.defaultProps.onClick).toHaveBeenCalledTimes(1);
  });
});
