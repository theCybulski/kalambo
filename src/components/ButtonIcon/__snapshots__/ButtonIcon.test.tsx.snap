// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon Renders properly with default props 1`] = `
<ButtonIcon
  icon="IconBrush.svg"
  isActive={false}
  onClick={[MockFunction]}
>
  <styled.button
    onClick={[MockFunction]}
  >
    <button
      className="sc-AxiKw fEooJD"
      onClick={[MockFunction]}
    >
      <styled.div
        icon="IconBrush.svg"
        isActive={false}
      >
        <div
          className="sc-AxjAm guzPJa"
        />
      </styled.div>
    </button>
  </styled.button>
</ButtonIcon>
`;
